test-plugins/supervising_provider_plugin/protocols/my_protocol.py

```diff
@@ -0,0 +1,46 @@
+import json
+
+from canvas_sdk.effects import Effect, EffectType
+from canvas_sdk.events import EventType
+from canvas_sdk.protocols import BaseProtocol
+from canvas_sdk.v1.data.staff import Staff
+from logger import log
+
+
+class Protocol(BaseProtocol):
+    """You should put a helpful description of this protocol's behavior here."""
+
+    RESPONDS_TO = [
+        EventType.Name(EventType.PRESCRIBE__SUPERVISING_PROVIDER__POST_SEARCH),
+        EventType.Name(EventType.PRESCRIBE__SUPERVISING_PROVIDER__PRE_SEARCH),
+        EventType.Name(EventType.REFILL__SUPERVISING_PROVIDER__POST_SEARCH),
+        EventType.Name(EventType.REFILL__SUPERVISING_PROVIDER__PRE_SEARCH),
+        EventType.Name(EventType.ADJUST_PRESCRIPTION__SUPERVISING_PROVIDER__POST_SEARCH),
+        EventType.Name(EventType.ADJUST_PRESCRIPTION__SUPERVISING_PROVIDER__PRE_SEARCH),
+    ]
+
+    NARRATIVE_STRING = "I was inserted from my supervising provider plugin's protocol."
+
+    def compute(self) -> list[Effect]:
+        """This method gets called when an event of the type RESPONDS_TO is fired."""
+        log.info(self.NARRATIVE_STRING)
+
+        results = self.context.get("results")
+
+        if results is None:
+            return [Effect(type=EffectType.AUTOCOMPLETE_SEARCH_RESULTS, payload=json.dumps(None))]
+
+        post_processed_results = []
+        for result in results:
+            staff = Staff.objects.get(dbid=result["value"])
```

Comment:
So we really want to avoid using `dbid`. I know we're using them in a lot of places, but going forward we're trying to standardize on the "id" column. 

In order to do that, you'll need to update your adapter to instead use the Staff's `key` as the value for the search.


Also, given you don't need the whole staff, you'd get a performance increase if you instead did `Staff.objects.values_list("spi_number", flat=True).get(whatever)`

Reply:
@nmpsilva the ask of this ticket was to add new events already using capabilities that exist. These events are live for other Autocomplete fields and the `result` in context only has `value` as the `dbid`. If we are changing the context of the event, that would be scope creep for GTM and should be handled in a new ticket. This plugin was to prove that the events are being called and annotations can be handled.